test/upgrade/upgrade_test.go

```diff
@@ -16,20 +16,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package upgrade
+package upgrade_test
```

Comment:
Why is this package changed? It's already under `test/upgrade` so isn't it redundant?

Reply:
It's a good practice in Go to write tests in the `_test` package. That disallow to test private functions, forcing all tests to use only public interfaces, and making code more testable in the end.

There's also a golangci-lint linter that enforce that https://github.com/maratori/testpackage#motivation

Reply:
OK. Thanks